runtime/tests/checker/runtimetype_test.go

```diff
@@ -0,0 +1,772 @@
+            `,
+			valid: true,
+		},
+		{
+			name: "resource",
+			code: `
+              resource R {}
+              let result = VariableSizedArrayType(Type<@R>())
+            `,
+			valid: true,
+		},
+		{
+			name: "type mismatch",
+			code: `
+              let result = VariableSizedArrayType(3)
+            `,
+			valid: false,
+		},
+		{
+			name: "too many args",
+			code: `
+              let result = VariableSizedArrayType(Type<Int>(), Type<Int>())
+            `,
+			valid: false,
+		},
+		{
+			name: "too few args",
+			code: `
+              let result = VariableSizedArrayType()
+            `,
+			valid: false,
+		},
+	}
+
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			checker, err := ParseAndCheck(t, testCase.code)
+
+			if testCase.valid {
+				require.NoError(t, err)
+				assert.Equal(t,
+					sema.MetaType,
+					RequireGlobalValue(t, checker.Elaboration, "result"),
+				)
+			} else {
+				require.Error(t, err)
+			}
+		})
+	}
+}
+
+func TestCheckConstantSizedArrayTypeConstructor(t *testing.T) {
+
+	t.Parallel()
+
+	cases := []struct {
+		name  string
+		code  string
+		valid bool
+	}{
+		{
+			name: "String",
+			code: `
+              let result = ConstantSizedArrayType(Type<String>(), 3)
+            `,
+			valid: true,
+		},
+		{
+			name: "Int",
+			code: `
+				let result = ConstantSizedArrayType(Type<Int>(), 2)
+            `,
+			valid: true,
+		},
+		{
+			name: "resource",
+			code: `
+              resource R {}
+              let result = ConstantSizedArrayType(Type<@R>(), 4)
+            `,
+			valid: true,
+		},
+		{
+			name: "type mismatch first arg",
+			code: `
+              let result = ConstantSizedArrayType(3, 4)
+            `,
+			valid: false,
+		},
+		{
+			name: "type mismatch second arg",
+			code: `
+              let result = ConstantSizedArrayType(Type<Int>(), "")
+            `,
+			valid: false,
+		},
+		{
+			name: "too many args",
+			code: `
+              let result = ConstantSizedArrayType(Type<Int>(), 3, 4)
+            `,
+			valid: false,
+		},
+		{
+			name: "one arg",
+			code: `
+              let result = ConstantSizedArrayType(Type<Int>())
+            `,
+			valid: false,
+		},
+		{
+			name: "no args",
+			code: `
+              let result = ConstantSizedArrayType()
+            `,
+			valid: false,
+		},
+	}
+
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			checker, err := ParseAndCheck(t, testCase.code)
+
+			if testCase.valid {
+				require.NoError(t, err)
+				assert.Equal(t,
+					sema.MetaType,
+					RequireGlobalValue(t, checker.Elaboration, "result"),
+				)
+			} else {
+				require.Error(t, err)
+			}
+		})
+	}
+}
+
+func TestCheckDictionaryTypeConstructor(t *testing.T) {
+
+	t.Parallel()
+
+	cases := []struct {
+		name  string
+		code  string
+		valid bool
+	}{
+		{
+			name: "String/Int",
+			code: `
+              let result = DictionaryType(Type<String>(), Type<Int>())
```

Comment:
Same here: Shouldn't the construction functions require argument labels here? 

It seems like the argument labels for the constructor functions are not checked